File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/repository/DepartmentRepository.java
package com.seclib.htbp.hosp.repository;

import com.seclib.htbp.model.hosp.Department;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface DepartmentRepository extends MongoRepository<Department,String> {
    Department getDepartmentByHoscodeAndDepcode(String hoscode, String depcode);

}

File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/repository/DepartmentRepository.kt
package com.seclib.htbp.hosp.repository

import org.springframework.data.mongodb.repository.MongoRepository
import com.seclib.htbp.model.hosp.Department
import org.springframework.stereotype.Repository

@Repository
interface DepartmentRepository : MongoRepository<Department?, String?> {
    fun getDepartmentByHoscodeAndDepcode(hoscode: String?, depcode: String?): Department?
}